servers/rendering/rendering_device.cpp

```diff
@@ -242,8 +335,57 @@ RID RenderingDevice::_uniform_set_create(const Array &p_uniforms, RID p_shader,
 	return uniform_set_create(uniforms, p_shader, p_shader_set);
 }
 
-Error RenderingDevice::_buffer_update(RID p_buffer, uint32_t p_offset, uint32_t p_size, const Vector<uint8_t> &p_data, uint32_t p_post_barrier) {
-	return buffer_update(p_buffer, p_offset, p_size, p_data.ptr(), p_post_barrier);
+Error RenderingDevice::buffer_update(RID p_buffer, uint32_t p_offset, uint32_t p_size_bytes, const Vector<uint8_t> &p_byte_data, uint32_t p_post_barrier) {
+	return buffer_update(p_buffer, p_offset, p_size_bytes, p_byte_data.ptr(), p_post_barrier);
+}
+
+Error RenderingDevice::buffer_update_generic(RID p_buffer, uint32_t p_index, uint32_t p_size, const Variant &p_data, uint32_t p_post_barrier) {
+	size_t elem_size = 0;
+	size_t array_size = 0;
+	const uint8_t *data = getPackedArrayData(p_data, elem_size, array_size);
+	ERR_FAIL_COND_V_MSG(data == nullptr, Error::ERR_INVALID_PARAMETER, "Can't update buffer with data of type " + Variant::get_type_name(p_data.get_type()) + ".");
+
+	return buffer_update(p_buffer, p_index * elem_size, p_size * elem_size, data, p_post_barrier);
+}
+
+template <typename T>
+Vector<T> RenderingDevice::buffer_get_data(RID p_buffer) {
+	Vector<T> buffer_data;
+	uint8_t *data = nullptr;
+	_buffer_get_data(p_buffer, &data, [&buffer_data](size_t p_size) {
```

Comment:
@reduz will have to decide if he wants to allow a lambda here. As far as I know, the general rule is not to use them, but reduz can approve on a case-by-case basis.

Reply:
I can shot a bullet in my own foot by saying that maybe we can have a method to retrieve the buffer size to be able to initialize the `void *` before calling `buffer_get_data`. I didn't check the feasibility of this solution.

Reply:
No lambda, just write to a void* and you are supposed to know the size and have pre allocated it.